diff --git a/uc_order/uc_order.info.inc b/uc_order/uc_order.info.inc
index 2e45576..e4f05c4 100644
--- a/uc_order/uc_order.info.inc
+++ b/uc_order/uc_order.info.inc
@@ -165,6 +165,7 @@ function uc_order_entity_property_info() {
           'label' => t('Products'),
           'description' => t('The products that have been ordered.'),
           'setter permission' => 'edit orders',
+          'getter callback' => 'uc_order_get_product_list',
           'clear' => array('order_total'),
         ),
       ),
diff --git a/uc_order/uc_order.module b/uc_order/uc_order.module
index f62ea56..b04efc3 100644
--- a/uc_order/uc_order.module
+++ b/uc_order/uc_order.module
@@ -1616,6 +1616,13 @@ function uc_order_get_product_count($order) {
 }
 
 /**
+ * Entity API getter for the list of order products.
+ */
+function uc_order_get_product_list($order) {
+  return array_keys($order->products);
+}
+
+/**
  * Determines if an order is shippable.
  *
  * An order can be shipped if any of its products can be shipped.
